src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchExceptionTranslator.java

```diff
@@ -16,25 +16,48 @@
 
 package org.springframework.data.elasticsearch.core;
 
+import sun.net.ConnectionResetException;
+
 import java.net.ConnectException;
 import java.util.List;
 
+import org.apache.http.ConnectionClosedException;
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.ElasticsearchStatusException;
+import org.elasticsearch.common.ValidationException;
 import org.springframework.dao.DataAccessException;
 import org.springframework.dao.DataAccessResourceFailureException;
+import org.springframework.dao.DataIntegrityViolationException;
 import org.springframework.dao.support.PersistenceExceptionTranslator;
 import org.springframework.data.elasticsearch.NoSuchIndexException;
+import org.springframework.data.elasticsearch.SpringDataElasticsearchException;
 import org.springframework.util.CollectionUtils;
 import org.springframework.util.ObjectUtils;
 import org.springframework.util.StringUtils;
 
 /**
  * @author Christoph Strobl
+ * @author Peter-Josef Meisch
  * @since 3.2
  */
 public class ElasticsearchExceptionTranslator implements PersistenceExceptionTranslator {
 
+	/**
+	 * translates a Throwable. Exceptions that are no {@link RuntimeException}s are wrapped in a RuntimeException
+	 * 
+	 * @param throwable the Throwable to map
+	 * @return the translated RuntimeException.
+	 * @since 4.0
+	 */
+	public RuntimeException translateException(Throwable throwable) {
```

Comment:
Let's stick with the `translateExceptionIfPossible` pattern that returns a potentially nullable `DataAccessException`. Bubbling of the original exception happens then on Template API level.

Reply:
this is code I just moved from [`ReactiveElasticsearchTemplate`](https://github.com/spring-projects/spring-data-elasticsearch/blob/3.2.x/src/main/java/org/springframework/data/elasticsearch/core/ReactiveElasticsearchTemplate.java#L776-L783).
If `translateException` would return null for an exception that is not mapped to a `DataAccessException`, then the reactive code would create a `MonoError(null)` which  will result in an NPE.
So for me the logic to try to translate the exception - and, if that cannot be done, return it wrapped in a `RuntimeException`if it is not already one, makes sense.

Reply:
Sorry, I missed to explain that callers of `translateExceptionIfPossible` are required to check for the exception translation outcome to prevent NPEs. Let's keep that method in the Template implementations for now.